Makefile

```diff
@@ -24,7 +24,7 @@ CLIENT_ARCH ?= $(shell go env GOARCH)
 
 ifneq ($(SKIP_DOCKER),true)
 	PROJECT_ROOT := $(dir $(realpath $(firstword $(MAKEFILE_LIST))))
-	GO_DEV_IMAGE := quay.io/deis/lightweight-docker-go:v0.7.0
+	GO_DEV_IMAGE := krancour/go-tools:v0.1.0
```

Comment:
What does this image provide beyond the official `golang` image?  I'm assuming handy utilities, also possibly with a slimmer footprint?

Although I'd place preference on using an official `golang` image, perhaps if we decide to go with this one, we can add a note and link to the source code somewhere in this repo?  Or, alternatively, we could house a Dockerfile "clone" of `krancour/go-tools` here in this repo, that would serve as the base image for all of the updated Dockerfiles as seen in this commit.  It may start out as a clone, but could potentially diverge as is needed by this project... and then the Dockerfile contents would be even easier to see/update/etc. WDYT?

Reply:
`lightweight-docker-go` has been used for a long time (since the Deis days) and has been periodically updated. It includes Go + Go-related tools commonly used (then and now). Things like dep, packr, linters, etc.

It was inconvenient for me to make further updates to that image (for reasons I can explain offline), so I started a new project that is exactly what I have described above, except newer versions of everything.

I am completely unopposed to moving this somewhere under brigadecore and, in fact, wanted to talk to you about that. It is the way it is now simply to unblock me, but I didn't imagine this to be the way it would be left. Perhaps I should have tagged this PR as WIP.

Reply:
Upon further reflection, I would be in favor of changing the name of this target to `resolve-dependencies`. It may be more verbose, but its function is crystal clear and does not (even coincidentally) hint at what the underlying dependency manager is.

Reply:
I'd be fine with a link for now to the Dockerfile that this image is based on in this suite of changes and then a follow-up to perhaps bring in said Dockerfile, either to this repo itself or perhaps a separate repo in the `brigadecore` org.  All in the name of visibility and open access to make updates/changes to this Dockerfile.

Reply:
The metadata in the Docker Hub already links to the Dockerfile in GitHub-- and the builds are automated directly via Docker Hub.

https://hub.docker.com/repository/docker/krancour/go-tools/general

Reply:
> then a follow-up to perhaps bring in said Dockerfile

Follow up sounds good, but can we make it its own top-level "utility" project in brigadecore since we're using it at across at least two other projects? Seems it would keep things simpler also if we don't have to build the build image at the start of the pipeline and can use a pre-built image that has all our tools.

Reply:
Yup ^^, I like it.